### ISSUE TITLE
NullPointerException with composed-schema containing oneOf in the components

### ISSUE THREAD

**Issue by jmini** [swagger-api/swagger-codegen-generators#15]
I have some trouble with a OpenAPI spec containing a `oneOf` switch. (from documentation [Data Models (Schemas) > oneOf, anyOf, allOf, not](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/))

When I try to generate the java client, I get an error caused by a null pointer Exception (version `1.0.0-SNAPSHOT`).

#### OpenAPI Spec:

```
openapi: 3.0.1
info:
  title: oneOf test
  version: '1.0'
servers:
  - url: 'http://localhost:8000/'
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Pet'
      responses:
        '200':
          description: Updated
components:
  schemas:
    Pet:
      oneOf:
        - $ref: '#/components/schemas/Cat'
        - $ref: '#/components/schemas/Dog'
    Dog:
      type: object
      properties:
        bark:
          type: boolean
        breed:
          type: string
          enum: [Dingo, Husky, Retriever, Shepherd]
    Cat:
      type: object
      properties:
        hunts:
          type: boolean
        age:
          type: integer
```

#### Stacktrace:

```
Exception in thread "main" java.lang.RuntimeException: Could not process model 'Pet'.Please make sure that your schema is correct!
	at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:361)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:726)
	at <my code>
Caused by: java.lang.NullPointerException
	at io.swagger.codegen.languages.DefaultCodegenConfig.fromModel(DefaultCodegenConfig.java:1210)
	at io.swagger.codegen.languages.java.AbstractJavaCodegen.fromModel(AbstractJavaCodegen.java:845)
	at io.swagger.codegen.DefaultGenerator.processModels(DefaultGenerator.java:959)
	at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:356)
	... 3 more
```

#### Note:

When instead of having a `Pet` schema in the components, the composed-schema is inlined in the requestBody (as in the [example](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/#oneof) from the docs), then the generator works.

```
openapi: 3.0.1
info:
  title: oneOf test
  version: '1.0'
servers:
  - url: 'http://localhost:8000/'
paths:
  /pets:
    patch:
      requestBody:
        content:
          application/json:
            schema:
              oneOf:
                - $ref: '#/components/schemas/Cat'
                - $ref: '#/components/schemas/Dog'
      responses:
        '200':
          description: Updated
components:
  schemas:
    Dog:
      type: object
      properties:
        bark:
          type: boolean
        breed:
          type: string
          enum: [Dingo, Husky, Retriever, Shepherd]
    Cat:
      type: object
      properties:
        hunts:
          type: boolean
        age:
          type: integer
```

**Comment by jmini** [comment]
The first step is to fix the NullPointerException.

Should it be fixed in [DefaultCodegenConfig (line 1210)](https://github.com/swagger-api/swagger-codegen-generators/blob/master/src/main/java/io/swagger/codegen/languages/DefaultCodegenConfig.java#L1210) by assuming the `composed.getAllOf()` can be null.
Or should `getAllOf()` in [ComposedSchema (line 38)](https://github.com/swagger-api/swagger-core/blob/2.0/modules/swagger-models/src/main/java/io/swagger/v3/oas/models/media/ComposedSchema.java#L38) be null safe (in this case the change should be made in the [swagger-core](https://github.com/swagger-api/swagger-core/) repository).

@HugoMario: have you an input for this question?

**Comment by jmini** [comment]
I have tried the second solution (fixing `ComposedSchema` directly in the core), but from a JSON point of view, `null` and empty list are not the same:

Current JSON:
```
type: object
anyOf:
- $ref: '#/components/schemas/UserObject'
- $ref: '#/components/schemas/EmployeeObject'
```

JSON after my "null-safe" modification in swagger-core:
```
type: object
allOf: []
anyOf:
- $ref: '#/components/schemas/UserObject'
- $ref: '#/components/schemas/EmployeeObject'
oneOf: []
```

This is why I have considered that it is better to fix the `NullPointerException` in this repository => Pull Request #16

**Comment by HugoMario** [comment]
agree

**Comment by jmini** [comment]
Thank you!

**Comment by srividhyak27** [comment]
Hi,
Is this fix available in 3.0.0-rc0  ?
I am having a similar issue where NullPointer exception is thrown, my schema uses "anyOf" .
Thanks,
Sri

**Comment by HugoMario** [comment]
hey @srividhyak27 , i think this fix is available for 3.0.0-rc1, can you please check and let me know?

**Comment by srividhyak27** [comment]
Hi @HugoMario ,  thanks,
this issue (NullPointerException when the schema has anyOf) seems to be fixed in 3.0.0-rc1.

But, in 3.0.0-rc1, there is an issue while referencing external yaml files containing schema.
java.lang.RuntimeException: Unable to load RELATIVE ref: ./yamls/test.yaml

can you please provide some pointers on how to address this ?

Thanks,
Srividhya
